internal/handlers/lokistack_create_or_update.go

```diff
@@ -102,12 +107,49 @@ func CreateOrUpdateLokiStack(ctx context.Context, req ctrl.Request, k k8s.Client
 			errCount++
 			continue
 		}
+
+		if !reflect.DeepEqual(op, crtlutil.OperationResultNone) {
+			collectMetrics = true
+		}
+
 		l.Info(fmt.Sprintf("Resource has been %s", op))
 	}
 
 	if errCount > 0 {
 		return kverrors.New("failed to configure lokistack resources", "name", req.NamespacedName)
 	}
 
+	if collectMetrics {
+		collectMetricsFor(&stack.Spec)
```

Comment:
The injection point here to call `collectMetricsFor` looks flawed to me by the fact that it runs once per reconciliation loop for the same `LokiStackSpec`. IUC this means that we increment the counters/gauges every time we reconcile an update. For example, assuming the operator owns only one single `LokiStack` resource, every time we update this resource the counter for `lokistack_scale_out_size_total` will go up by one. However, this counter should stay to one as there is only one `LokiStack` resource for the given size.